src/reducers/entities/posts.js

```diff
@@ -39,7 +39,10 @@ function handleReceivedPost(posts = {}, postsInChannel = {}, postsInThread = {},
         ];
     }
 
-    return {posts: nextPosts, postsInChannel: nextPostsInChannel, postsInThread: nextPostsInThread};
+    const combineSystemPosts = combineSystemPost(nextPostsInChannel[channelId], nextPosts);
```

Comment:
When I looked at this yesterday, I wasn't really sure about combining system posts in the store. I was thinking it would be better to have the redux store match what is on the server, but then we'd have to remember to combine system posts and such all over the place before showing them in the UI, so I think this actually ends up being a really elegant solution to this